mlflow/protos/service.proto

```diff
@@ -12,9 +12,28 @@ option (scalapb.options) = {
 };
 
 service MlflowService {
+
+
+  // Get metadata for an experiment.
+  // This endpoint will return deleted experiments.
```

Comment:
I feel like this should be active only by default, much like list experiments

Reply:
@akarloff thanks for the comment, the goal here was actually to have similar behavior to the GetExperiment REST API, which does return deleted experiments by default. It seems GetExperimentByName has more use cases in common with GetExperiment (fetching an experiment based on unique ID / name) than with ListExperiments, and it'd be good to avoid the confusion that would arise from having the two experiment-getting endpoints behave differently.

Reply:
@smurching Clarifying the behavior of the REST API as per your last comment is a decent work around. Supporting reuse of experiment names after deletion is something I would ultimately like to see, and could be supported with the documentation you proposed.

Reply:
Sounds good, I'll add that clarification - thanks again for the input!